### ISSUE TITLE
Enhancement: Directed Acyclic Graph Layout

### ISSUE THREAD

**Issue by tawheeler** [JuliaGraphs/NetworkLayout.jl#20]
NetworkLayout.jl currently has the Buchheim algorithm for laying out trees. Often data sets are hierarchical, but are not trees. It'd be great to have a layout algorithm for directed acyclic graphs, such as [the Sugiyama method](https://en.wikipedia.org/wiki/Layered_graph_drawing).

**Comment by oxinabox** [comment]
I am preparing to release https://github.com/oxinabox/LayeredLayouts.jl which does have algorithms for DAG layout.
It currently just has one, but its a good one (provably optimal in number of crossings).
Basic docs should be up tonight

**Comment by SimonDanisch** [comment]
Any reason to not put it under the same API under NetworkLayout?

**Comment by simonschoelly** [comment]
I wonder how much NetworkLayout.jl is still maintained. Apparently it is still a dependency of GraphRecipes.jl, so it must still work but the current situation is:
- the latest commit there is two years old
- there is a lot of code that is duplicated in GraphPlot.jl

Given that GraphPlot.jl is also not in the best state, I wonder if some effort should be put into NetworkLayout.jl so that it is usually from multiple graph plotting packages.

**Comment by SimonDanisch** [comment]
Yeah, that's the idea^^ I'm simply not aware of anyone using it, and nobody has been complaining about anything, that's why there hasn't been any commits...

**Comment by oxinabox** [comment]
> Any reason to not put it under the same API under NetworkLayout?

It has *huge* dependencies. JuMP, ECOS, and Cbc.

**Comment by SimonDanisch** [comment]
Has it?
![image](https://user-images.githubusercontent.com/1010467/96349933-09878980-10b3-11eb-9f8d-f2d2ba6c0f7d.png)
Am I missing something?

**Comment by SimonDanisch** [comment]
Ah, lol, I guess you mean the other package :D
